pac4j-oauth/src/main/java/org/pac4j/oauth/profile/generic/GenericOAuth20Profile.java

```diff
@@ -0,0 +1,89 @@
+package org.pac4j.oauth.profile.generic;
+
+import org.pac4j.core.context.Pac4jConstants;
+import org.pac4j.core.profile.Gender;
+import org.pac4j.oauth.profile.OAuth20Profile;
+
+import java.util.Locale;
+
+/**
+ * <p>This is the user profile for the generic OAuth 2.0 client.</p>
+ * <p>It is returned by the {@link org.pac4j.oauth.client.GenericOAuth20Client}  </p>
+ *
+ * @author aherrick
+ * @since 1.9.2
+ */
+public class GenericOAuth20Profile extends OAuth20Profile {
+
+    private transient GenericAttributesDefinition attributes = new GenericAttributesDefinition();
```

Comment:
I think I would make the `attributesDefinition` `final static` and attached to this profile. If you have a new profile, you'll have a new `attributesDefinition` and the problem is how to define what profile is returned by what client (and it's a more general issue than just for OAuth).